hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -350,6 +368,63 @@ public void createDbStore(MutableVolumeSet dbVolumeSet) throws IOException {
     }
   }
 
+  /**
+   * Ensure that volume is initialized properly with
+   * cleanup path. Should disk be re-inserted into
+   * cluster, cleanup path should already be on
+   * disk. This method syncs the HddsVolume
+   * with the path on disk.
+   * 
+   * @param id clusterId or scmId
+   */
+  public void checkTmpDirPaths(String id) {
+    Path tmpPath = createTmpPath(id);
+    deleteServiceDirPath = tmpPath.resolve(TMP_DELETE_SERVICE_DIR);
+  }
+
+  private Path createTmpPath(String id) {
+
+    // HddsVolume root directory path
+    String hddsRoot = getHddsRootDir().toString();
+
+    // HddsVolume path
+    String vol = HddsVolumeUtil.getHddsRoot(hddsRoot);
+
+    Path volPath = Paths.get(vol);
+    Path idPath = Paths.get(id);
```

Comment:
The id is the <clusterid / scmid> based on the version.  It may be best to check and set the `id` from `VersionDatanodeFeatures.chooseContainerPathID` here.  Once set here, all calls to the tmp dir container delete service path can be independent of the version, having properly set the path correctly in one place.

Reply:
I've been looking at all the usage and [this](https://github.com/apache/ozone/blob/f291058fa8974fe3a2844df5d27ec444e9826742/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/StorageVolumeUtil.java#L241-L265) seems the only case where clusterID might not be available and we need to set it. 

For calling `VersionDatanodeFeatures.ScmHA.chooseContainerPathID` we need to know clusterId which might not be available in the corner case we are trying to cover here.

It makes more sense to me to overload `checkTmpDirPaths()` and have a separate `checkTmpDirPaths(String id)` where we choose the id in StorageVolumeUtil and pass it as a parameter.  

@neils-dev What do you think?

Reply:
We should be able to use `chooseContainerPathID(StorageVolume volume, String clusterID) `inside the original `checkTmpDirPaths` so one place sets the proper location for the subdir of hdds.  The `chooseContainerPathID `should resolve the proper location of the hdds subdir independent of version.  Inside `checkTmpDirPaths`, just check if id is null, if so pass an emptyString for `clusterID `to `chooseContainerPathId `and it should return the proper location of the hdds subdir (scmid or clusterid).  Let me know how it goes.

Reply:
This is handled in the new method `getIdDir()`. I'll resolve this conversation.